synapse/_scripts/synapse_port_db.py

```diff
@@ -637,6 +656,21 @@ async def run(self) -> None:
                 allow_outdated_version=True,
             )
 
+            # For safety, ensure auto_vacuums are disabled.
+            if await self.sqlite_store.db_pool.runInteraction(
+                "is_sqlite_autovacuum_enabled", self._is_sqlite_autovacuum_enabled
+            ):
+                end_error = (
+                    "auto_vacuum is enabled in the SQLite database."
+                    " (This is not the default configuration.)\n"
+                    " This script relies on rowids being consistent and must not"
+                    " be used if the database could be vacuumed between re-runs.\n"
+                    " To disable auto_vacuum, you need to stop Synapse and run the following SQL:\n"
+                    " PRAGMA auto_vacuum=off;\n"
+                    " VACUUM;"
```

Comment:
Why do you need to vacuum here after disabling autovacuum? To ensure there is no vacuum in progress when the script runs?

Reply:
You can't change the `auto_vacuum` setting without `VACUUM`ing if there are existing tables

Reply:
> You can't change the `auto_vacuum` setting without `VACUUM`ing if there are existing tables

Does that mean the vacuum should happen first?

Reply:
No; you set the PRAGMA then VACUUM to make it take effect.